gradle-spoon-plugin/src/main/kotlin/com/jaredsburrows/spoon/SpoonTask.kt

```diff
@@ -18,96 +15,86 @@ import java.time.Duration
 open class SpoonTask : DefaultTask() { // tasks can't be final
 
   /** Results baseOutputDir. */
-  @get:OutputDirectory var outputDir: File
+  @get:OutputDirectory lateinit var outputDir: File
+  @Internal lateinit var buildDir: File
 
-  /** Variant of the test */
-  @Internal lateinit var variant: TestVariant
+  /** For testing only. */
+  @Internal var testing: Boolean = false
 
-  /** Application APK (eg. app-debug.apk). */
-  private lateinit var applicationApk: File
+  /** Spoon Extension. */
+  @Internal lateinit var spoonExtension: SpoonExtension
+
+  /** Android SDK directory */
+  @Internal lateinit var sdkDirectory: File
 
   /** Instrumentation APK (eg. app-debug-androidTest.apk). */
-  private lateinit var instrumentationApk: File
+  @Internal lateinit var instrumentationApk: File
```

Comment:
You could make those properties `internal` instead of `public` to reduce the exposed API
```suggestion
  @Internal internal lateinit var instrumentationApk: File
```

Reply:
Thanks!

Reply:
I am seeing:
```
      Please refer to https://docs.gradle.org/7.4.2/userguide/validation_problems.html#missing_annotation for more details about this problem.
    - Error: Type 'com.jaredsburrows.spoon.SpoonTask' property 'variantName$gradle_spoon_plugin' is missing an input or output annotation.
      
      Reason: A property without annotation isn't considered during up-to-date checking.
      
      Possible solutions:
        1. Add an input or output annotation.
        2. Mark it as @Internal.
```

Reply:
Ah sorry I missed that. I guess you need to use `@get:Internal` to annotate the getter instead of the Kotlin property.